src/util/prolog/prologSession.ts

```diff
@@ -0,0 +1,90 @@
+import { format_answer, type } from "tau-prolog/modules/core";
+
+export class PrologSession {
+
+		public readonly session: any;
+
+		constructor(program: string, limit?: number) {
+				this.session = new type.Session(limit);
+				this.session.consult(program);
+		}
+
+		// Options
+		// 0 means return all values (boolean and variable values)
+		// 1 means only return boolean returns
+		// 2 means only return variable values
+		public async executeQuery(query: string, option: number = 0): Promise<string[][]> {
```

Comment:
## Multiple pieces of advise here
### Types
Typescript supports both enums and union-types. Both are more expressive than bare `number`s. 

Example for union type:
```typescript
public async executeQuery(query: string, filter?: "booleans" | "variables"): ...
```
Of course you can define the union type separately as well:
```typescript
(export) type FilterSetting = "booleans" | "variables";
...
public async executeQuery(query: string, filter?: FilterSetting): ...
```

Example for enum type:
```typescript
(export) enum FilterSetting {
  None,
  Boolean,
  Variables,
}
...
public async executeQuery(query: string, filter: FilterSetting): ...
```

### Style
When a function does different things based on a parameter (or even returns different types!) it's good to split the function up or at least make it private and offer public specific wrappers to make it easier to use.